src/Utility/Hash.php

```diff
@@ -370,7 +365,9 @@ protected static function _simpleOp($op, $data, $path, $values = null)
                 }
             } elseif ($op === 'remove') {
                 if ($i === $last) {
-                    unset($_list[$key]);
+                    if (!is_string($_list)) {
```

Comment:
Should we check for other scalar values? Or ensure that `$_list` is an array before unsetting from it?